examples/js/loaders/GLTFLoader.js

```diff
@@ -2444,13 +2465,7 @@ THREE.GLTFLoader = ( function () {
 				if ( primitive.extensions && primitive.extensions[ EXTENSIONS.KHR_DRACO_MESH_COMPRESSION ] ) {
 
 					// Use DRACO geometry if available
-					geometryPromise = extensions[ EXTENSIONS.KHR_DRACO_MESH_COMPRESSION ]
-						.decodePrimitive( primitive, parser )
-						.then( function ( geometry ) {
-
-							return addPrimitiveAttributes( geometry, primitive, parser );
-
-						} );
+					geometryPromise = createDracoPrimitive( primitive );
```

Comment:
This was the main issue. `var` is not block scoped, unlike `let`, so by the time the callback returns, the value of `primitive` will have changed. All primitives were loaded using the last primitive definition. 😓

https://codeburst.io/asynchronous-code-inside-an-array-loop-c5d704006c99

Reply:
😁